File: app/src/main/java/com/hover/stax/permissions/StaxPermissionDialog.java
package com.hover.stax.permissions;

import android.animation.ObjectAnimator;
import android.app.Activity;
import android.view.LayoutInflater;
import android.view.animation.DecelerateInterpolator;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.hover.stax.R;
import com.hover.stax.utils.Utils;
import com.hover.stax.views.StaxDialog;

public class StaxPermissionDialog extends StaxDialog {
	private final static String TAG = "StaxPermissionDialog";

	public final static int OVERVIEW = 0, OVERLAY = 1, ACCESS = 2;
	private int cmd;

	public StaxPermissionDialog(@NonNull Activity a) {
		this(a, a.getLayoutInflater());
	}

	private StaxPermissionDialog(Activity a, LayoutInflater inflater) {
		super(a);
		context = a;
		view = inflater.inflate(R.layout.stax_permission_dialog, null);
		((TextView) view.findViewById(R.id.app_name)).setText(Utils.getAppName(a));
		customNegListener = null;
		customPosListener = null;
	}

	public void animateProgressTo(int percent) {
		ObjectAnimator oa = ObjectAnimator.ofInt(view.findViewById(R.id.progress_indicator), "progress", percent);
		oa.setDuration(800);
		oa.setInterpolator(new DecelerateInterpolator());
		oa.start();
	}
}

File: app/src/main/java/com/hover/stax/permissions/StaxPermissionDialog.kt
/*
 * Copyright 2022 Stax
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.hover.stax.permissions

import android.animation.ObjectAnimator
import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.animation.DecelerateInterpolator
import android.widget.TextView
import com.hover.stax.R
import com.hover.stax.views.StaxDialog

class StaxPermissionDialog private constructor(a: Activity, inflater: LayoutInflater) : StaxDialog(a) {

    constructor(a: Activity) : this(a, a.layoutInflater)

    init {
        ctx = a
        mView = inflater.inflate(R.layout.dialog_permission, null)
        customPosListener = null
        customNegListener = null
    }

    fun animateProgressTo(percent: Int) {
        val animator = ObjectAnimator.ofInt(mView.findViewById(R.id.progress_indicator), "progress", percent)
        animator.apply {
            duration = 800
            interpolator = DecelerateInterpolator()
            start()
        }
    }

    fun setPath(pathStringRes: Int): StaxDialog {
        mView.findViewById<TextView>(R.id.path_text)?.let {
            it.visibility = View.VISIBLE
            it.text = ctx.getString(pathStringRes)
        }
        return this
    }

    fun setHelperIcon(iconRes: Int): StaxDialog {
        mView.findViewById<TextView>(R.id.perm_message)?.let {
            it.setCompoundDrawablesWithIntrinsicBounds(iconRes, 0, 0, 0)
        }
        return this
    }
}